### DISCUSSION TITLE
Idea: drag drop of url with supported 3d model creates sample script

### DISCUSSION THREAD

**Discussion by hrgdavor** [hrgdavor/jscadui#52] [Ideas]
jscad app reacts to drag and drop of an url .... we then create a sample script that shows how to read model from ulr and how to start manipulating it

```js
import { translate } from '@jscad/modeling'
import { deserialize } from '@jscad/io'

export const main = async () => {
const object = await deserialize('https://raw.githubusercontent.com/jscad/OpenJSCAD.org/master/packages/io/x3d-deserializer/tests/ElevationGrids.x3d')
  
  // you can do some jscad operations on the loaded object and return it
  let out = translate([0,0,0],object)
  return out
}
 ```

we can later expand on this by providing a wizard with questions to be able to end-up with a script that is useful. (we can find few common example what somebody would want to do with a loaded model and guide user to producing starting script)

**Comment by z3dev** [comment]
It depends on what the user wants. Some users are looking for the JSCAD script from the deserializer, while others just want to import the mesh and manipulate it (above example). It would be cool if the UI supported such an option.

**Comment by hrgdavor** [comment]
That is the idea actually, when someone drops a stl file, a modal dialog shows-up asking questions what would user like to do with the model, and mentioned examples would be on the list along with others we might come up.
